src/shared/trim-and-dedent.ts

```diff
@@ -71,24 +67,5 @@ export function trimAndDedent(
   strings: TemplateStringsArray,
   ...params: any[]
 ): string {
-  if (strings.length < 2) {
-    // Trim leading and trailing empty lines from first (and only) string
-    const trimmedLines = [
-      trimLeadingEmptyLines(trimTrailingEmptyLines(strings[0])),
-    ];
-    return dedent(trimmedLines, ...params);
-  }
-
-  const firstString = strings[0];
-  const lastString = strings[strings.length - 1];
-  const middle = strings.slice(1, strings.length - 1);
-
-  // Trim empty leading lines from first string, and empty trailing lines from last one
-  const trimmedLines = [
-    trimLeadingEmptyLines(firstString),
-    ...middle,
-    trimTrailingEmptyLines(lastString),
-  ];
-
-  return dedent(trimmedLines, ...params);
+  return trimLeadingAndTrailingEmptyLines(dedent([...strings], ...params));
```

Comment:
By dedenting first and then removing empty lines in the whole "resolved" string, we make sure that, if the very first or very last lines of the template string were actually params, empty lines on them are also removed.

The best thing about this change is that it simplifies the logic a lot.